packages/flutter_tools/test/commands/build_test.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/base/common.dart';
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/commands/build.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/version.dart';
+import 'package:flutter_tools/src/runner/flutter_command.dart';
+
+import '../src/common.dart';
+import '../src/context.dart';
+
+
+void main() {
+  group('Master channel warning', () {
+    testUsingContext('warns on master', () async {
+      final MockBuildCommand buildCommand = MockBuildCommand();
+      try {
+        await createTestCommandRunner(buildCommand).run(<String>['build', 'test']);
+      } on ToolExit catch(e) {}
+      expect(testLogger.statusText, contains('🐉 This is the master channel. Shipping apps from this channel is not recommended as it has not\n been as heavily tested as the stable channel. To build using the stable channel, consider using:\n    flutter channel stable'));
+      Cache.releaseLockEarly();
+    }, overrides: <Type, Generator>{
+      FlutterVersion: () => MockVersion('master'),
+    });
+
+    testUsingContext('no warning on stable', () async {
+      final MockBuildCommand buildCommand = MockBuildCommand();
+      try {
+        await createTestCommandRunner(buildCommand).run(<String>['build', 'test']);
+      } on ToolExit catch(e) {}
+      expect(testLogger.statusText, '');
+      Cache.releaseLockEarly();
+    }, overrides: <Type, Generator>{
+      FlutterVersion: () => MockVersion('stable'),
+    });
```

Comment:
Up to you, but you could write:

```dart
for (String channel in <String>['stable', 'dev', 'beta']) {
  testUsingContext('no warning on $channel', () async {
    ...
  }, overrides: <Type, Generator>{
    FlutterVersion: () => MockVersion(channel),
  });
}
```

I suspect the end result is less readable, so entirely fine and maybe better as-is.

Reply:
I think in terms of tests failing, this way, each case gets it own title and it will be easier to figure out what is wrong if it breaks. Otherwise, we may be trying to debug the iterations in the loop when, say, only the dev case is broken.

Reply:
That way actually should too, but as I say, it's pretty unreadable ;)